my_enumerable.rb

```diff
@@ -0,0 +1,196 @@
+        end
+      else
+        my_each do |value|
+        break result = true if value == args end
+      end
+    end
+
+    if !block_given? 
+      if is_a? Hash
+        my_each do |item|
+        break result = true if item[1] == true end
+      else
+        my_each do |item|
+        break result = true if item == true end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        break result = true if yield(item, value) == true end
+      else
+        my_each do |value|
+          break result = true if yield(value) == true  end
+      end
+    end
+    result
+  end
+
+  def my_all?(args = nil)
+    result = true
+    unless args.nil?
+      if is_a? Hash
+        my_each do |_item, value|
+        break result = false unless value.class == args end
+      elsif args.is_a? Regexp
+        my_each do |value| 
+          result = false  if (value =~ args).nil?        
+        end
+      elsif (args == String) || (args == Integer) ||  (args == Float) ||  (args == Numeric)
+        my_each do |value| 
+          break result = false unless value.class == args  
+        end
+      else
+        my_each do |value|
+        break result = false unless value == args end
+      end
+    end
+    unless block_given? && args.nil?
+      if is_a? Hash
+        my_each do |item|
+        result = false if item[1] == false || item[1].nil? end
+      else
+        my_each do |item|
+        result = false if item == false || item.nil? end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        break result = false if yield(item, value) == false || yield(item, value).nil? end
+      else
+        my_each do |value|
+          break result = false if yield(value) == false || yield(value).nil?
+        end
+      end
+    end
+    result
+  end
+
+  def my_none?(args = nil)
+    result = true
+    unless args.nil?
+     if is_a? Hash
+        my_each do |_item, value|
+        break result = false unless value.class == args end
+     elsif args.is_a? Regexp
+        my_each do |value| 
+          break result = false  unless (value =~ args).nil?        
+        end
+     elsif (args == String) || (args == Integer) ||  (args == Float) ||  (args == Numeric)
+        my_each do |value| 
+          break result = false if value.class == args  
+        end
+      else
+        my_each do |value|
+        break result = false if value == args end
+      end
+    end
+    unless block_given? && args.nil?
+      if is_a? Hash
+        my_each do |item|
+        result = false if item[1] == true end
+      else
+        my_each do |item|
+        result = false if item == true end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        break result = false if yield(item, value) == true end
+      else
+        my_each do |value|
+          break result = false if yield(value) == true
+        end
+      end
+    end
+    result
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?